### PR TITLE
Add Cancel Tx RpcApi

### PR DESCRIPTION
**Pull request by Ashuaidehao**

### CODE REVIEW

File: src/RpcServer/RpcServer.Wallet.cs [hunk 1]
```diff
@@ -319,6 +319,60 @@ protected virtual JToken SendToAddress(JArray _params)
...
+                Signers = signers,
+                Attributes = conflict,
+                Witnesses = Array.Empty<Witness>(),
+            };
+
+            try
+            {
+                using ScriptBuilder scriptBuilder = new();
+                scriptBuilder.Emit(OpCode.RET);
+                tx = wallet.MakeTransaction(system.StoreView, scriptBuilder.ToArray(), signers[0].Account, signers, conflict);
```
Comment (cschuchardt88) [inline src/RpcServer/RpcServer.Wallet.cs:351]:
```suggestion
                tx = wallet.MakeTransaction(system.StoreView, new[] { (byte)OpCode.RET }, signers[0].Account, signers, conflict);
```

**Review by cschuchardt88 — COMMENTED** [review]
Note: that any `RpcException` will need to be changed after #815 is merged, unless this is merged before it.

**Comment by Liaojinghui** [comment]
Might take a million years to merge. No need to wait for that.

File: src/RpcServer/RpcServer.Wallet.cs [hunk 2]
```diff
@@ -319,6 +319,58 @@ protected virtual JToken SendToAddress(JArray _params)
...
+            CheckWallet();
+            var txid = UInt256.Parse(_params[0].AsString());
+            TransactionState state = NativeContract.Ledger.GetTransactionState(system.StoreView, txid);
+            if (state != null)
+            {
+                throw new RpcException(32700, "This tx is already confirmed, can't be cancelled.");
+            }
+
+            var conflict = new TransactionAttribute[] { new Conflicts() { Hash = txid } };
+            Signer[] signers = _params.Count >= 2 ? ((JArray)_params[1]).Select(j => new Signer() { Account = UInt160.Parse(j.AsString()), Scopes = WitnessScope.CalledByEntry }).ToArray() : Array.Empty<Signer>();
```
Comment (superboyiii) [inline src/RpcServer/RpcServer.Wallet.cs:334]:
```suggestion
            Signer[] signers = _params.Count >= 2 ? ((JArray)_params[1]).Select(j => new Signer() { Account = AddressToScriptHash(j.AsString(), system.Settings.AddressVersion), Scopes = WitnessScope.CalledByEntry }).ToArray() : Array.Empty<Signer>();
```
To accept both UInt160 or Base58 Address

File: src/RpcServer/RpcServer.Wallet.cs [hunk 3]
```diff
@@ -319,6 +319,58 @@ protected virtual JToken SendToAddress(JArray _params)
...
+            }
+            else if (_params.Count >= 3)
+            {
+                var extracFee = _params[2].AsString();
+                AssetDescriptor descriptor = new(system.StoreView, system.Settings, NativeContract.GAS.Hash);
+                if (!BigDecimal.TryParse(extracFee, descriptor.Decimals, out BigDecimal decimalExtraFee) || decimalExtraFee.Sign <= 0)
+                {
+                    throw new RpcException(32702, "Incorrect Amount Format");
+                }
+                tx.NetworkFee += (long)decimalExtraFee.Value;
```
Comment (superboyiii) [inline src/RpcServer/RpcServer.Wallet.cs:368]:
```suggestion
                var extraFee = _params[2].AsString();
                AssetDescriptor descriptor = new(system.StoreView, system.Settings, NativeContract.GAS.Hash);
                if (!BigDecimal.TryParse(extraFee, descriptor.Decimals, out BigDecimal decimalExtraFee) || decimalExtraFee.Sign <= 0)
                {
                    throw new RpcException(32702, "Incorrect Amount Format");
                }
                tx.NetworkFee += (long)decimalExtraFee.Value;
```

**Review by superboyiii — DISMISSED** [review]

File: src/RpcServer/RpcServer.Wallet.cs [hunk 4]
```diff
@@ -319,6 +319,57 @@ protected virtual JToken SendToAddress(JArray _params)
...
+            CheckWallet();
+            var txid = UInt256.Parse(_params[0].AsString());
+            TransactionState state = NativeContract.Ledger.GetTransactionState(system.StoreView, txid);
+            if (state != null)
+            {
+                throw new RpcException(32700, "This tx is already confirmed, can't be cancelled.");
+            }
+
+            var conflict = new TransactionAttribute[] { new Conflicts() { Hash = txid } };
+            Signer[] signers = _params.Count >= 2 ? ((JArray)_params[1]).Select(j => new Signer() { Account = AddressToScriptHash(j.AsString(), system.Settings.AddressVersion), Scopes = WitnessScope.CalledByEntry }).ToArray() : Array.Empty<Signer>();
```
Comment (shargon) [inline src/RpcServer/RpcServer.Wallet.cs:334]:
```suggestion
            Signer[] signers = _params.Count >= 2 ? ((JArray)_params[1]).Select(j => new Signer() { Account = AddressToScriptHash(j.AsString(), system.Settings.AddressVersion), Scopes = WitnessScope.None }).ToArray() : Array.Empty<Signer>();
```

**Review by shargon — APPROVED** [review]

**Review by superboyiii — APPROVED** [review]

**Pull request merged**